pkg/generate/code/set_resource.go

```diff
@@ -1229,6 +1238,29 @@ func SetResourceForStruct(
 	for memberIndex, memberName := range sourceShape.MemberNames() {
 		targetMemberShapeRef := targetShape.MemberRefs[memberName]
 		if targetMemberShapeRef == nil {
+			// This can happen when the targetShape is a primitive
+			// and the sourceShape is a struct like when EC2 resource DHCPOptions
+			// needs to set NewDhcpConfiguration.Values (targetShape = string) field from
+			// DhcpConfiguration.Values (sourceShape = AttributeValue). AttributeValue
+			// has a 'Value' member whereas the string shape does not. However,
+			// there is a possibility that the targetShape can be set from
+			// a different member within sourceShape using SetConfig.
+			if currentField, ok := r.Fields[currentPath]; ok {
+				setCfg := currentField.GetSetterConfig(model.OpTypeList)
```

Comment:
I actually don't see how passing `model.OpTypeList` hard-coded here is going to work...

Reply:
Can you advise what should be done given `opType` is required for this api? I modeled after the similar behavior [here](https://github.com/aws-controllers-k8s/code-generator/blob/2eb18c5a67e5955149a6bb1eaa92440cdfed31e3/pkg/generate/ack/controller.go#L112).

Pass another parameter, opType, to setResource funcs?

Reply:
> I actually don't see how passing model.OpTypeList hard-coded here is going to work...

This only works if `SetterConfig` applies to ALL methods or just `model.OpTypeList`. Either way-- the gap you pointed out is still there (what if on `Create` only).